src/v2/manifest/mod.rs

```diff
@@ -286,7 +286,21 @@ fn evaluate_media_type(
     }
 }
 
-fn build_accept_headers() -> header::HeaderMap {
+fn build_accept_headers(registry: &str) -> header::HeaderMap {
+    if registry.ends_with("://gcr.io") || registry.ends_with(".gcr.io") {
```

Comment:
I'm happy to learn why you chose these conditions.

I've got a hunch that we can just check if it ends with `gcr.io` here, as the TLD is very unlikely to be overriden in a local DNS system, and if so, they would probably know what they're doing.

Reply:
well, I had following cases in mind:
"http://gcr.io" - true
"https://eu.gcr.io" - true
"http://my-super-gcr.io" - false